Library/MapleBacon/MapleBacon/ImageManager.swift

```diff
@@ -8,58 +8,66 @@ public class ImageManager {
-                }
-            }
+            return nil
+        }
+        
+        if nil == downloadsInProgress[url] {
+            let downloadOperation = ImageDownloadOperation(imageURL: url)
+            downloadOperation.qualityOfService = .userInitiated
+            downloadOperation.completionHandler = downloadHandlerWithStorage(url: url, cacheScaled: cacheScaled,
+                    imageView: imageView, storage: storage, completion: completion)
+            downloadsInProgress[url] = downloadOperation
+            downloadQueue.addOperation(downloadOperation)
+            return downloadOperation
         }
+        
+        completion?(ImageInstance(image: nil, state: .Downloading, url: nil), nil)
+        delay(delay: 0.1) {
+            _ = self.downloadImage(atUrl: url, cacheScaled: cacheScaled, imageView: imageView, storage: storage, completion: completion)
+        }
+
         return nil
     }
 
-    private func downloadHandlerWithStorage(url: NSURL, cacheScaled: Bool, imageView: UIImageView?, storage: Storage, completion: ImageDownloaderCompletion?) -> ImageDownloaderCompletion {
+    private func downloadHandlerWithStorage(url: URL, cacheScaled: Bool, imageView: UIImageView?, storage: Storage, completion: ImageDownloaderCompletion?) -> ImageDownloaderCompletion {
+        
         return { [weak self] (imageInstance, _) in
             self?.downloadsInProgress[url] = nil
             if let newImage = imageInstance?.image {
                 if cacheScaled && imageView != nil && newImage.images == nil {
-                    self?.resizeAndStoreImage(newImage, imageView: imageView!, storage: storage,
-                        key: url.absoluteString)
+                    self?.resizeAndStoreImage(image: newImage, imageView: imageView!, storage: storage,
+                        key: url.absoluteString!)
                 } else if let imageData = imageInstance?.data {
-                    storage.storeImage(newImage, data: imageData, forKey: url.absoluteString)
+                    //storage.storeImage(newImage, data: imageData, forKey: url.absoluteString)
+                    storage.store(data: imageData, forKey: url.absoluteString!)
                 }
                 completion?(ImageInstance(image: newImage, state: .New, url: imageInstance?.url), nil)
             }
         }
     }
 
     private func resizeAndStoreImage(image: UIImage, imageView: UIImageView, storage: Storage, key: String) {
-        Resizer.resizeImage(image, contentMode: imageView.contentMode, toSize: imageView.bounds.size,
-                interpolationQuality: .Default) {
-            resizedImage in
-            storage.storeImage(resizedImage, data: nil, forKey: key)
+        
+        let resizer = Resizer(image: image)
+        resizer.resize(toSize: imageView.bounds.size, contentMode: imageView.contentMode, interpolationQuality: CGInterpolationQuality.default) {
+            (resizedImage) in
```

Comment:
here the same as above with the closure parameters, same line as the opening brace